combined_iterator.go

```diff
@@ -31,28 +33,62 @@ type combinedIterator struct {
 }
 
 type combinedIteratorConfig struct {
-	snapshotConfig snapshotIteratorConfig
-	cdcConfig      cdcIteratorConfig
+	db                    *sqlx.DB
+	tableKeys             common.TableKeys
+	fetchSize             int
+	startSnapshotPosition *common.SnapshotPosition
+	startCdcPosition      *common.CdcPosition
+	database              string
+	tables                []string
+	serverID              common.ServerID
+	mysqlConfig           *mysqldriver.Config
+	disableCanalLogging   bool
 }
 
 func newCombinedIterator(
 	ctx context.Context,
 	config combinedIteratorConfig,
 ) (common.Iterator, error) {
-	iterator := &combinedIterator{}
-	var err error
-
-	iterator.cdcIterator, err = newCdcIterator(ctx, config.cdcConfig)
+	cdcIterator, err := newCdcIterator(ctx, cdcIteratorConfig{
+		tables:              config.tables,
+		mysqlConfig:         config.mysqlConfig,
+		tableKeys:           config.tableKeys,
+		disableCanalLogging: config.disableCanalLogging,
+		db:                  config.db,
+		startPosition:       config.startCdcPosition,
+	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to create cdc iterator: %w", err)
 	}
 
-	iterator.snapshotIterator, err = newSnapshotIterator(ctx, config.snapshotConfig)
+	if config.startCdcPosition == nil {
+		if err := cdcIterator.obtainStartPosition(ctx); err != nil {
```

Comment:
@alarbada I wanted to note that the, the lock in this function, shouldn't be released until the snapshotter has acquired the min/max of the tables.

Reply:
@lyuboxa Latest commits should address this (plus some cleanup), could you take one last look?